pallets/system/src/lib.rs

```diff
@@ -1428,7 +1456,12 @@ impl<T: Config> Pallet<T> {
 		BlockWeight::<T>::mutate(|current_weight| {
 			current_weight.set(weight, DispatchClass::Normal)
 		});
-		AllExtrinsicsLen::<T>::put(len as u32);
+		let block_length = Self::block_length();
+		let padded = kate::padded_len(len as u32, block_length.chunk_size());
```

Comment:
@MiguelDD1 might be a naive question, but why do we need the padded length here, if the padded block size is only used for calculating commitments?

Reply:
This is due to a security issue. 
We update the value here and **specially** at `check_weight.rs` (fn `do_pre_dispatch`). This latest one is called before every TX is executed, so It allows us to filter TXs just before add it to the block, and it means we are sure that `fn finalize()` is infallible.
That was the root-cause of  `BlockTooBig` issue: `fn finalize` MUST be infallible but the padding len limit was checked inside, having no way to recovery neither to pre-filter TX.